### PR TITLE
Implement Management VRF (Phase 2)

### LINKED ISSUE #60: Implement VRF generator for management VRF
## Summary

Implement the VRF generator that creates the management VRF and assigns interfaces to it.

## Background

Phase 2 of VyOS Router v3 adds Management VRF support. The models and parser already handle `ETHx_VROUTER_MANAGEMENT=YES` - this issue implements the generator that produces VyOS commands.

## Design Decisions (from implementation plan)

- VRF name: `management` (hardcoded)
- Table ID: `100` (hardcoded)
- Multiple interfaces supported

## Implementation

Create `src/vyos_onecontext/generators/vrf.py`:

1. **VRF creation**: `set vrf name management table 100`
2. **Interface assignment**: `set interfaces ethernet ethX vrf management` for each management interface
3. **VRF default route**: `set vrf name management protocols static route 0.0.0.0/0 next-hop <gw>`
   - Use same gateway selection algorithm as main routing (lowest-numbered interface with gateway != IP)

## VyOS Commands Generated

```vyos
set vrf name management table 100
set interfaces ethernet eth0 vrf management
set vrf name management protocols static route 0.0.0.0/0 next-hop 10.0.0.254
```

## Acceptance Criteria

- [ ] VRF only created if management interfaces exist
- [ ] All management interfaces assigned to VRF
- [ ] VRF default route uses correct gateway selection logic
- [ ] No VRF commands when no management interfaces

## References

- Phase 2 in deployment repo: `docs/docs/projects/active/vyos-router-v3/implementation-plan.md`
- VyOS VRF docs: https://docs.vyos.io/en/sagitta/configuration/vrf/index.html

---
Generated with [Claude Code](https://claude.ai/code) (Opus 4.5)

### LINKED ISSUE #61: Implement SSH service generator with VRF support
## Summary

Implement an extensible service generator pattern, starting with SSH VRF binding.

## Background

When management VRF is active, services like SSH should bind to that VRF for out-of-band management access. This issue creates the service generator infrastructure.

## Design

Create `src/vyos_onecontext/generators/service.py` with an extensible pattern:

```python
class SshServiceGenerator(BaseGenerator):
    """Bind SSH to management VRF if management interfaces exist."""
```

Future services (NTP, SNMP, etc.) can follow the same pattern.

## VyOS Commands Generated

```vyos
set service ssh vrf management
```

## Behavior

- If any interface has `management=True`: bind SSH to management VRF
- If no management interfaces: no SSH VRF commands (SSH uses default/global)

## Acceptance Criteria

- [ ] SSH bound to management VRF when management interfaces exist
- [ ] No SSH VRF commands when no management interfaces
- [ ] Code structured for easy addition of other VRF-aware services

## References

- Depends on #60 (VRF generator)
- VyOS SSH docs: https://docs.vyos.io/en/sagitta/configuration/service/ssh.html

---
Generated with [Claude Code](https://claude.ai/code) (Opus 4.5)

### LINKED ISSUE #62: Integrate VRF and service generators into command pipeline
## Summary

Modify `generators/__init__.py` to integrate the new VRF and service generators in the correct order.

## Background

VyOS commands have ordering dependencies. The VRF must exist before interfaces can be assigned to it, and before SSH can bind to it.

## Implementation

Update `src/vyos_onecontext/generators/__init__.py`:

1. Import `VrfGenerator` and `SshServiceGenerator`
2. Add to `generate_config()` in correct order
3. Update `__all__` exports

## Required Command Order

```
1. set system host-name <hostname>
2. set system login user vyos authentication public-keys...
3. set vrf name management table 100              # VRF creation
4. set interfaces ethernet ethX vrf management    # VRF assignment
5. set vrf name management protocols static route...  # VRF routing
6. set interfaces ethernet ethY address <cidr>    # All interface addresses
7. set interfaces ethernet ethY mtu <mtu>
8. set service ssh vrf management                 # SSH VRF binding
9. set protocols static route 0.0.0.0/0 next-hop  # Main VRF routing
```

## Acceptance Criteria

- [ ] VRF generator called before interface generator
- [ ] SSH service generator called after VRF generator
- [ ] Main routing generator remains last
- [ ] All generators exported in `__all__`

## References

- Depends on #60 (VRF generator) and #61 (SSH service generator)

---
Generated with [Claude Code](https://claude.ai/code) (Opus 4.5)

### LINKED ISSUE #63: Add unit tests for VRF and service generators
## Summary

Add comprehensive unit tests for the VRF and SSH service generators.

## Implementation

Create `tests/test_vrf_generators.py` with test cases for both generators.

## Test Cases: VrfGenerator

| Test | Description |
|------|-------------|
| `test_no_management_interfaces` | No VRF commands when no management interfaces |
| `test_single_management_interface` | VRF created with single management interface |
| `test_multiple_management_interfaces` | Multiple interfaces join VRF, default route uses lowest-numbered |
| `test_mixed_management_and_data` | Only management interfaces join VRF |
| `test_management_gateway_equals_ip_skipped` | Gateway == IP skipped for VRF default route |
| `test_management_no_gateway` | No default route when no gateway specified |

## Test Cases: SshServiceGenerator

| Test | Description |
|------|-------------|
| `test_no_management_vrf` | No SSH VRF binding when no management interfaces |
| `test_ssh_bound_to_management_vrf` | SSH bound to VRF when management interface exists |

## Integration Tests

| Test | Description |
|------|-------------|
| `test_management_vrf_full_config` | Complete config with management VRF |
| `test_command_order_vrf_before_ssh` | VRF creation before SSH binding |

## Acceptance Criteria

- [ ] All unit tests pass
- [ ] Tests cover edge cases (no gateway, gateway == IP, mixed interfaces)
- [ ] Integration tests verify command ordering

## References

- Depends on #60, #61, #62

---
Generated with [Claude Code](https://claude.ai/code) (Opus 4.5)

### LINKED ISSUE #64: Update documentation for Management VRF (Phase 2)
## Summary

Update documentation to reflect Management VRF implementation.

### context-reference.md

Add/update section on VRF behavior:
- Document `ETHx_VROUTER_MANAGEMENT=YES` effect
- Explain SSH accessibility change when management VRF is active
- Note that multiple interfaces can join management VRF

### design.md

Update feature status table:
- Change "Management VRF" from "Model ready, generator pending" to "**Implemented**"

### implementation-plan.md (deployment repo)

- Check off Phase 2 tasks (2.1, 2.2, 2.3)
- Mark documentation checkpoint complete

## Acceptance Criteria

- [ ] VRF behavior documented in context-reference.md
- [ ] Feature status updated in design.md
- [ ] Phase 2 marked complete in implementation-plan.md

## References

- Depends on #60, #61, #62, #63 (all implementation complete)
- Phase 2 checkpoint in deployment repo

---
Generated with [Claude Code](https://claude.ai/code) (Opus 4.5)

### PR DESCRIPTION
**Pull request by duplico**
## Summary

Implements Phase 2 of VyOS Router v3 contextualization: Management VRF support.

- Add VRF generator for management VRF creation and interface assignment
- Add SSH service generator with VRF binding
- Update documentation with VRF behavior

### New Files
- `src/vyos_onecontext/generators/vrf.py` - VRF generator
- `src/vyos_onecontext/generators/service.py` - SSH service generator

### Modified Files
- `src/vyos_onecontext/generators/__init__.py` - Integration
- `tests/test_generators.py` - 17 new tests
- `docs/context-reference.md` - VRF behavior documentation
- `docs/design.md` - Feature status update

## VyOS Commands Generated

When `ETHx_VROUTER_MANAGEMENT=YES`:
```vyos
set vrf name management table 100
set interfaces ethernet ethX vrf management
set vrf name management protocols static route 0.0.0.0/0 next-hop <gateway>
set service ssh vrf management
```

## Test plan

- [x] All 298 tests pass
- [x] VRF only created when management interfaces exist
- [x] SSH bound to VRF when management interfaces exist
- [x] Correct command ordering (VRF before SSH binding)
- [x] Gateway selection logic works for management VRF

Fixes #60, fixes #61, fixes #62, fixes #63, fixes #64

🤖 Generated with [Claude Code](https://claude.ai/code) (Opus 4.5)

### CODE REVIEW

File: src/vyos_onecontext/generators/vrf.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        """
+
+        def natural_sort_key(iface: InterfaceConfig) -> float:
+            """Extract numeric portion of interface name for natural sorting.
+
+            Ensures eth2 sorts before eth10 (numeric order, not lexicographic).
+            Non-eth interfaces are sorted after all numbered eth interfaces.
+            """
+            match = re.match(r"eth(\d+)", iface.name)
+            return float(match.group(1)) if match else float("inf")
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/vyos_onecontext/generators/vrf.py]: Fixed in 85cf3c4 - extracted natural_sort_key to generators/utils.py and updated both routing.py and vrf.py to import from there.

File: tests/test_generators.py [hunk 2]
```diff
@@ -679,3 +683,434 @@ def test_generate_config_command_order_with_routing(self):
...
+                    gateway=IPv4Address("192.168.1.254"),
+                    management=True,
+                ),
+            ],
+        )
+        commands = generate_config(config)
+
+        # Find indices of different command types
+        interface_idx = next(
+            i for i, cmd in enumerate(commands) if "interfaces ethernet" in cmd
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in 85cf3c4 - updated the test pattern to use 'if "interfaces ethernet" in cmd and " address " in cmd' for more specific matching of interface address assignment commands.
